src/main/java/edu/oakland/production/database/DatabasePersistentStorageClass.java

```diff
@@ -0,0 +1,167 @@
+package edu.oakland.production.database;
+
+import edu.oakland.helper.admin.LocationDataPoint;
+import edu.oakland.helper.admin.TrackData;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class DatabasePersistentStorageClass {
+
+  int rfidData;
+  ArrayList<LocationDataPoint> locDataPoint = new ArrayList<LocationDataPoint>();
+  ArrayList<TrackData> trackData = new ArrayList<TrackData>();
+  String[] nextSatellites;
+  String mode;
+
+  /**
+   * Creates the the class for managing the persistent storage.
+   * Provides rest of Database with functions.
+   *
+   * @param rfidData the rfid integer.
+   * @param locDataPoint the location data point array list to initialize.
+   * @param trackData the track data  array list to initialize.
+   * @param nextSatellites the string array of satellites.
+   */
+  public DatabasePersistentStorageClass(int rfidData, 
+      ArrayList<LocationDataPoint> locDataPoint, 
+      ArrayList<TrackData> trackData, 
+      String[] nextSatellites) {
+
+    this.rfidData = rfidData;
+    this.locDataPoint = locDataPoint;
+    this.trackData = trackData;
+    this.nextSatellites = nextSatellites;
+    this.mode = "normal";
+  }
+
+
+  /**
+   * Returns the rfidData integer.
+   * 
+   *
+   * 
+   * 
+   * 
+   * 
+   */
+
+  public int locateRfidData() {
+    return rfidData;
+  }
+
+  /**
+   * Returns a locationpoint point.
+   * after providing the offset.
+   *
+   * @param offset the offset (int) of the data needed.
+   */
+
+  public LocationDataPoint getLocationDataPoint(int offset) {
+
+    // If trackdata has size of length 4 , then the offset can not be > 3 so we use size-1
+    if (offset > this.locDataPoint.size() - 1) {
+      // throw new IndexOutOfBoundsException("Offset " + offset + " is out of bounds!");
+      return null;
+    }
+
+    return this.locDataPoint.get(offset);
+  }
+
+  /**
+   * Takes a location data point and stores it.
+   *
+   *
+   * @param locationDataPoint the locationDataPoint data type.
+   */
+
+  public void storeLocationDataPoint(LocationDataPoint locationDataPoint) {
+    if (locationDataPoint == null) {
+      throw new IllegalArgumentException("locationDataPoint cannot be null");
+    }
+
+    this.locDataPoint.add(locationDataPoint);
+  }
+
+  /**
+   * Retrieve track data.
+   *
+   *
+   * @param offset the offset (int) of the track data needed.
+   */
+
+  public TrackData getTrackData(int offset) {
+    // If we have a td of L=4, then the offset can't be >3 so we use length-1
+    if (offset > this.trackData.size() - 1) {
+      return null;
```

Comment:
Thoughts on returning an empty `TrackData`? This would greatly reduce the amount of issues Middleware_02 would have processing this output, as they always need to set the message. If might be better to do something like this (note: we would need to change the tests to make this change, but it will probable reduce an edge case later on).
```suggestion
      return new TrackData(new LocationDataPoint[0]);
```